### PR TITLE
Specify exact version for dependencies + using swiftlint plugin to autofix

### PR DESCRIPTION
**Pull request by Muhieddine-El-Kaissi**
- Pinning exact version of dependencies
- Using swiftlint auto fix plugin:

Screenshot 2023-05-19 at 10 55 45 AM

### CODE REVIEW

**Review by scottasoutherland — COMMENTED** [review]
Ins't pinning dependencies to exact versions a bad practice ? Then it means any automatic updates won't work since you'll have to go identify all the dependencies you can upgrade ?

**Review by dwroth — COMMENTED** [review]
It looks like you are also removing a bunch of other packages (e.g. sourcekitten, swift-argument-parser, swiftsyntax, ...).  Is that intentional?

File: Package.swift [hunk 1]
```diff
@@ -17,12 +17,12 @@ let package = Package(
...
-        .package(url: "https://github.com/thumbtack/TTCalendarPicker.git", from: "0.2.0"),
-        .package(url: "https://github.com/pointfreeco/swift-snapshot-testing.git", from: "1.11.0"),
-        .package(url: "https://github.com/thumbtack/thumbprint-tokens.git", from: "13.0.1"),
-        .package(url: "https://github.com/realm/SwiftLint.git", from: "0.51.0"),
-        .package(url: "https://github.com/nicklockwood/SwiftFormat", from: "0.50.4"),
+        .package(url: "https://github.com/SnapKit/SnapKit.git", exact: "5.6.0"),
+        .package(url: "https://github.com/thumbtack/TTCalendarPicker.git", exact: "0.2.0"),
+        .package(url: "https://github.com/pointfreeco/swift-snapshot-testing.git", exact: "1.11.0"),
+        .package(url: "https://github.com/thumbtack/thumbprint-tokens.git", exact: "13.0.1"),
+        .package(url: "https://github.com/lukepistrol/SwiftLintPlugin", exact: "0.2.6"),
```
Comment (scottasoutherland) [inline Package.swift:24]: Why was this changed ? This github page points you to the official one from realm ?

**Comment by Muhieddine-El-Kaissi** [comment]
> It looks like you are also removing a bunch of other packages (e.g. sourcekitten, swift-argument-parser, swiftsyntax, ...). Is that intentional?

there were dependencies automatically resolved by xcode. it seems that they are not needed anymore with the newer versions of lint/format

**Comment by Muhieddine-El-Kaissi** [comment]
> Ins't pinning dependencies to exact versions a bad practice ? Then it means any automatic updates won't work since you'll have to go identify all the dependencies you need to upgrade first?

we don't need to update automatically to a newer version in case there are issues. in the main app we are pinning all libraries to the exact version + xcode won't have to go to check for a newer version to fetch and resolve
when we pin a library to a specific version, xcode will resolve and update all its dependencies automatically

**Comment by scottasoutherland** [comment]
> we don't need to update automatically to a newer version in case there are issues.

I understand we don't want to always be updating. But lets say that library X version 1.0 depends on library Y version 1.0. And we pin both to 1.0. Then we try to update library X to 1.1, it will fail because library Y is pinned to 1.0 , whereas if we pinned both to their latest major version, updating one would be able to resolve and update the other

**Comment by Muhieddine-El-Kaissi** [comment]
@scottasoutherland updated .thanks!

File: Package.swift [hunk 2]
```diff
@@ -17,12 +17,12 @@ let package = Package(
...
     dependencies: [
-        .package(url: "https://github.com/SnapKit/SnapKit.git", .upToNextMajor(from: "5.6.0")),
+        .package(url: "https://github.com/SnapKit/SnapKit.git", from: "5.6.0"),
         .package(url: "https://github.com/thumbtack/TTCalendarPicker.git", from: "0.2.0"),
         .package(url: "https://github.com/pointfreeco/swift-snapshot-testing.git", from: "1.11.0"),
         .package(url: "https://github.com/thumbtack/thumbprint-tokens.git", from: "13.0.1"),
-        .package(url: "https://github.com/realm/SwiftLint.git", from: "0.51.0"),
-        .package(url: "https://github.com/nicklockwood/SwiftFormat", from: "0.50.4"),
+        .package(url: "https://github.com/lukepistrol/SwiftLintPlugin", from: "0.2.6"),
+        .package(url: "https://github.com/nicklockwood/SwiftFormat", from: "0.51.9"),
```
Comment (scottasoutherland) [inline Package.swift:25]: Did you mean to bump this to 51.9 , i see you changing it *back* to 51.8 in other PRs ?

**Review by scottasoutherland — APPROVED** [review]

File: Package.swift (hunk 2 above)
Comment (Muhieddine-El-Kaissi) [reply Package.swift:25]: yes, I will bump to 0.51.9 in here and I will do the same in the main app too
